README.md

```diff
@@ -1,3 +1,131 @@
+        .git
+        addon.json
+        aldryn_config.py
+        settings.json
+        setup.py
+        my_addon
+            __init__.py
+            models.py
+            ...
+private
+    sass
+        ...
+static
+    js
+    css
+    img
+templates
+    base.html
+.git
+.gitignore
+Dockerfile
+manage.py
+Procfile
+README.md
+requirements.in
+requirements.txt
+settings.py
+urls.py
+wsgi.py
+```
+
+## ``addons`` and ``addons-dev``
+
+There is a folder here for every Addon that is installed in the project and 
+contains these 3 files:
+
+* ``aldryn_addon.py``: holds the configuration logic and the form to edit exposed settings (just a copy of the one from the Addon package)
+* ``addon.json``: holds some metadata about the Addon (just a copy of the one from the Addon package)
+* ``settings.json`` the (non secret) values for the form fields defined in ``aldryn_addon.py``
+
+These files (and ``requirements.in``) will be updated/commited by the
+controlpanel whenever an Addon is added or upgraded over the UI.
+
+If you're developing an Addon locally, you'd just clone the Addon git repo
+into ``addons-dev`` and ``pip install -e`` it.
+``aldryn_addon.py`` and ``addon.json`` are already at the right location.
+Remember to ``.gitignore`` ``settings.json`` in the Addon repo.
+
+For this prototype the ``addons-dev`` folder actually contains the sourcecode
+within the git repo. In real usage the code for each Addon would be in a git repo of its own
+and manually cloned into the ``addons-dev`` folder.
+
+Even *Django* itself is an Addon. It constructs all the common settings like
+``DATABASES``, ``CACHES`` and many more. It also provides the manage.py script
+and a standardised command to start the webserver, encapsulating all the 
+nitty-gritty details. What is also planned, is that Addons will be able to
+register scripts for the ``migrate`` step (mainly used for database
+migrations and fix-mptt by django-cms).
+So upgrading the common ``aldryn-django`` package can also
+upgrade the way django is started and how migrations are run (Django 1.6 to
+Django 1.7 migrations upgrade anyone? ;-) ).
+ 
+Similarly *django CMS* is an Addon. It adds all the django-cms settings. In 
+the future it will be able to register migrate steps as well
+(e.g delete orphaned plugins).
+In the first iteration the django-cms Addon will be fat with a lot of other
+dependencies built in (django-filer, aldryn-boilerplates, ...). Later, when
+we have Addon dependency support, we'll split it into smaller Addons.
+
+
+## settings.py
+
+Here you can add any project specific settings. The special thing about it, is
+that we mostly get the settings from the installed Addons:
+
+```
+import aldryn_addons.settings
+aldryn_addons.settings.load(locals())
+```
+
+But you can do whatever you want in here. Including shooting yourself in the
+foot. The controlpanel will auto update the ``INSTALLED_ADDONS`` setting based on what
+is done in the UI.
+
```

Comment:
I wonder if there is a good way to show user what standard Django settings values do we use by default and what override with this project template. So that he knows if it is safe/supported to set/override something.

Also, is there some plan to support project template updating (for fixes, for example)?